baseapp/baseapp.go

```diff
@@ -510,6 +513,19 @@ func handleQueryCustom(app *BaseApp, path []string, req abci.RequestQuery) (res
 	}
 }
 
+func (app *BaseApp) validateHeight(req abci.RequestBeginBlock) error {
+	if req.Header.Height < 0 {
```

Comment:
I think Tendermint starts at height 1 now.

Reply:
True, but can the same be said for other consensus engines?

Reply:
Not necessarily, but I think we should panic otherwise, because elsewhere in the SDK we assume that the first block height is 1.

Reply:
Fair point -- updated.

Reply:
Can a comment be made somewhere or we amend ABCI to stipulate that the first block is at height 1?

Reply:
@ValarDragon yes, I'll follow up on this.